gpflow/models/model.py

```diff
@@ -57,8 +52,29 @@ def log_prior_density(self) -> tf.Tensor:
         else:
             return to_default_float(0.0)
 
+    def log_posterior_density(self, *args, **kwargs) -> tf.Tensor:
+        """
+        This may be the posterior with respect to the hyperparameters (e.g. for
+        GPR) or the posterior with respect to the function (e.g. for GPMC and
+        SGPMC). It assumes that maximum_likelihood_objective() is defined
+        sensibly.
+        """
+        return self.maximum_likelihood_objective(*args, **kwargs) + self.log_prior_density()
+
+    def maximum_a_posteriori_objective(self, *args, **kwargs) -> tf.Tensor:
+        """
+        Maximum a-posteriori objective. Assumes that maximum_likelihood_objective() is the log marginal likelihood and adds the log density of all priors.
+        """
+        return self.maximum_likelihood_objective(*args, **kwargs) + self.log_prior_density()
```

Comment:
remind me why there are two of these with the same default implementation

Reply:
I wanted to call it _training_loss and have it contain the minus sign already, but I think in reviewing my other proposal (which got subsequently merged in here) that wasn't deemed acceptable, so we called it something else ?

Reply:
SGPMC and GPMC are the only cases where they're overridden, but there these methods differ by `self.log_prior_density()`. Are these defaults correct?

Reply:
They don't differ?

Reply:
If you think we can simply call it _training_loss here I'd honestly prefer that.

Reply:
why wasn't it deemed acceptable? i'm generally in favour of making everything and anything private, so `_training_loss` sounds promising